### ISSUE TITLE
Untyped additional properties not available via SDK

### ISSUE THREAD

**Issue by rabii17** [pulumi/pulumi-azure-native#2668]
### What happened?

notify-keyspace-events setting is not available in the redis configuration [arguments](https://www.pulumi.com/registry/packages/azure-native/api-docs/cache/redis/#rediscommonpropertiesredisconfiguration).
We are using Pulumi operator to manage Azure Cache for Redis.

### Expected Behavior

notify-keyspace-events can be set using Pulumi since there is an Azure API for that.

### Steps to reproduce

Add the `notifyKeyspaceEvents` in the `redisConfiguration`

### Output of `pulumi about`

`Existing properties are: rdbBackupFrequency, aofBackupEnabled, maxmemoryReserved, authnotrequired, maxmemoryDelta and 9 others\n    pulumi:pulumi:Stack mbcs  (0.86s) 7 messages\n\nDiagnostics:\n  pulumi:pulumi:Stack (mbcs):\n    \u001b[31mError\u001b[0m: Property notifyKeyspaceEvents does not exist on 'azure-native:cache:RedisCommonPropertiesRedisConfiguration'\n`

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by danielrbradley** [comment]
Hi @rabii17 thanks for getting in touch.

So, the `RedisConfiguration` description from the specification seems to hint that this property should be settable, but it doesn't appear in the SDK. 

Here's the relavant part of the specification: 

https://github.com/Azure/azure-rest-api-specs/blob/b5f54222a4a925eb0239cd6653ab8b27185195f4/specification/redis/resource-manager/Microsoft.Cache/stable/2023-04-01/redis.json#L1898-L1975

The list of properties in the specification does not include `notify-keyspace-events`. The `additionalProperties` is present except it's missing the type information of the additional properties required for us to also allow passing in a raw dictionary of values (similar to `tenantSettings`. 

I think what we should be doing here is that if additionalProperties is specified without a type, we should just allow a dictionary of Objects to be passed instead of the typed object.

**Comment by rabii17** [comment]
Hello @danielrbradley,

Thank you for your quick response.

I was checking  `RedisCommonPropertiesRedisConfigurationArgs` in the python SDK as an example [here](https://github.com/pulumi/pulumi-azure-native/blob/0294216a6dd216e844837a99b9950b690d63b0bf/sdk/python/pulumi_azure_native/cache/_inputs.py#L443) and those additional properties are missing.
So allowing a dictionary there, should solve the problem as you suggested.

**Comment by danielrbradley** [comment]
@rabii17 A second workaround you can look at for the time being is to use [the transformation resource option](https://www.pulumi.com/docs/concepts/options/transformations/) to dynamically modify the input object to add the additional fields.

**Comment by rabii17** [comment]
Hello @danielrbradley,
Unfortunately, the `transformation` is not an option for us since we are using Pulumi CRs (YAML) to create the resources

**Comment by scarylooking** [comment]
I'm having the exact same issue but with `aad-enabled`. The Azure REST API spec does not mention it (even in the comment for `additionalProperties`) but it is the only way to enable RBAC access to an Azure Redis Cache. It also looks like there may be other properties set via `additionalProperties` too. 

https://learn.microsoft.com/en-us/rest/api/redis/redis/create?tabs=HTTP#redisconfiguration

https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/azure-cache-for-redis/cache-azure-active-directory-for-authentication.md

**Comment by mikhailshilkov** [comment]
Duplicate of https://github.com/pulumi/pulumi-azure-native/issues/2668, let's track there

**Comment by rasmusjuhl** [comment]
> I'm having the exact same issue but with `aad-enabled`. The Azure REST API spec does not mention it (even in the comment for `additionalProperties`) but it is the only way to enable RBAC access to an Azure Redis Cache. It also looks like there may be other properties set via `additionalProperties` too.
> 
> https://learn.microsoft.com/en-us/rest/api/redis/redis/create?tabs=HTTP#redisconfiguration
> 
> https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/azure-cache-for-redis/cache-azure-active-directory-for-authentication.md

We are also seeing this issue. Any news?
